_posts/2022-05-23-deploy-agent-on-fedora-coreos.markdown

```diff
@@ -0,0 +1,314 @@
+      contents:
+        local: files/key.pem
+      user:
+        id: 1000
+    - path: /etc/pki/consumer/cert.pem
+      mode: 0666
+      contents:
+        local: files/cert.pem
+      user:
+        id: 1000
+    - path: /etc/pki/consumer/ca.pem
+      mode: 0666
+      contents:
+        local: files/ca.pem
+      user:
+        id: 1000
+    - path: /etc/yggdrasil/config.toml
+      mode: 0666
+      contents:
+        local: files/config.toml
+    # Dont log audit messages
+    - path: /etc/sysctl.d/20-silence-audit.conf
+      mode: 0644
+      contents:
+        inline: |
+          # Raise console message logging level from DEBUG (7) to WARNING (4)
+          # to hide audit messages from the interactive console.
+          kernel.printk=4
+    # Set operator hostname
+    - path: /etc/hosts
+      overwrite: true
+      contents:
+        inline: |
+          IPADDRESS DOMAIN
+    # linger root
+    - path: /etc/systemd/logind.conf
+      overwrite: true
+      contents:
+        inline: |
+          KillExcludeUsers=root
+    # disable selinux
+    - path: /etc/selinux/config
+      overwrite: true
+      contents:
+        inline: |
+          SELINUX=disabled
+          SELINUXTYPE=targeted
+```
+
+First, we need to copy our custom services to `systemd` folder. This is done under `trees` section.
+The section `directories` creates the `/etc/pki/customer` folder where we will copy the certificates. Next, we copy the certificates and the configuration file from the host to vm.
+The modification of `20-silence-audit.conf` is not mandatory but it will be useful not to have those debug messages in the log.
+
+A couple more files and we are done. First we need to setup `Flotta API` address and domain in the `/etc/hosts` file. 
+Next, we are enabling `linger` for `root` and disabling `SELINUX`.
+
+Voilà! We are done with the `butane` file. 
+
+All that is rest is arrange all these file like this:
+```
+.
+├── files
+│   ├── ca.pem
+│   ├── cert.pem
+│   ├── config.toml
+│   └── key.pem
+├── spec.bu
+└── systemd
+    ├── flotta-setup.service
+    └── flotta-start.service
+
+```
+
+Here we put all the files to be copied to OS under `files` folder and our two services under `systemd`. 
+
+### Generate ignition file
+
+The ignition file can be generate either with `podman` or directly by installing `butane`:
+
+##### Via podman
+
+```bash
+podman run --interactive --rm quay.io/coreos/butane:release \
+       --pretty --strict spec.bu spec.ign
+```
+
+##### Via butane
+
+Butane is available as Fedora package:
+```bash
+sudo dnf install -y butane
+```
+
+Run butane on the Butane file:
+```bash
+butane --pretty --strict spec.bu > spec.ign
+```
+
+### Ignite CoreOS instance
+
+For this blog, I use `libvirt` but you have lots of other solution to create you instance. Please take a look at CoreOS [doc](https://docs.fedoraproject.org/en-US/fedora-coreos/).
+
+After we fetch the latest image for `qemu`, the instance can be lunched using:
```

Comment:
In the future it would be great to provide steps how to build rpm ostree update with flotta. It would simplify the configuration.